### ISSUE TITLE
Cannot read property 'end' of undefined when click stop debug button

### ISSUE THREAD

**Issue by lyquocnam** [Dart-Code/Dart-Code#1155]
this happen when i click stop debug button.
```
Cannot read property 'end' of undefined
```
![image](https://user-images.githubusercontent.com/8347826/44177025-57d92080-a116-11e8-8d9f-27253c350c33.png)

**Comment by DanTup** [comment]
Are you debugging a Dart VM or Flutter app? Can you check the dev console (Help -> Toggle Developer Console) and see if there's a stack trace for the error?

**Comment by lyquocnam** [comment]
i work with flutter, and this is the console log:
```
 INFO no standard startup: not just one text editor
console.ts:136 [Extension Host] vscode-icons is active!
notificationsAlerts.ts:39 TypeError: Cannot read property 'end' of undefined
    at t.stopSession (debugAdapter.ts:242)
    at t.stopServer (rawDebugSession.ts:501)
    at rawDebugSession.ts:492
    at Object.g [as _notify] (winjs.base.js:1191)
    at Object.enter (winjs.base.js:867)
    at n.Class.derive._oncancel._run (winjs.base.js:1089)
    at n.Class.derive._oncancel._completed (winjs.base.js:1057)
    at rawDebugSession.ts:226
    at t.e.acceptMessage (debugAdapter.ts:117)
    at t.handleData (debugAdapter.ts:189)
    at Socket.<anonymous> (debugAdapter.ts:155)
    at emitOne (events.js:116)
    at Socket.emit (events.js:211)
    at addChunk (_stream_readable.js:263)
    at readableAddChunk (_stream_readable.js:250)
    at Socket.Readable.push (_stream_readable.js:208)
    at TCP.onread (net.js:594)
t.onDidNotificationChange @ notificationsAlerts.ts:39
(anonymous) @ notificationsAlerts.ts:27
e.fire @ event.ts:140
t.notify @ notifications.ts:113
t.error @ notificationService.ts:52
(anonymous) @ debugActionsWidget.ts:125
e.fire @ event.ts:140
(anonymous) @ actions.ts:221
v @ winjs.base.js:1230
enter @ winjs.base.js:867
_run @ winjs.base.js:1089
_completed @ winjs.base.js:1057
(anonymous) @ rawDebugSession.ts:226
e.acceptMessage @ debugAdapter.ts:117
t.handleData @ debugAdapter.ts:189
(anonymous) @ debugAdapter.ts:155
emitOne @ events.js:116
emit @ events.js:211
addChunk @ _stream_readable.js:263
readableAddChunk @ _stream_readable.js:250
Readable.push @ _stream_readable.js:208
onread @ net.js:594
log.ts:171   ERR custom request failed: Error: custom request failed
    at file:///C:/Users/lynam/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:3151:972
    at Object.g [as _notify] (file:///C:/Users/lynam/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:151:729)
    at Object.enter (file:///C:/Users/lynam/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:155:568)
    at n.Class.derive._oncancel._run (file:///C:/Users/lynam/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:156:902)
    at n.Class.derive._oncancel._error (file:///C:/Users/lynam/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:156:419)
    at n.Class.derive._oncancel.t.internalSend [as _oncancel] (file:///C:/Users/lynam/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:2574:839)
    at n.Class.derive._oncancel._cancelAction (file:///C:/Users/lynam/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:158:865)
    at Object.enter (file:///C:/Users/lynam/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:154:547)
    at n.Class.derive._oncancel._run (file:///C:/Users/lynam/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:156:902)
    at n.Class.derive._oncancel.cancel (file:///C:/Users/lynam/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:156:31)
    at n.Class.derive._creator._cancelAction (file:///C:/Users/lynam/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:157:175)
    at Object.enter (file:///C:/Users/lynam/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:154:547)
    at n.Class.derive._creator._run (file:///C:/Users/lynam/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:156:902)
    at n.Class.derive._creator.cancel (file:///C:/Users/lynam/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:156:31)
    at file:///C:/Users/lynam/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:2579:800
    at Array.forEach (<anonymous>)
    at file:///C:/Users/lynam/AppData/Local/Programs/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:2579:768
notificationsAlerts.ts:39 TypeError: Cannot read property 'end' of undefined
    at t.stopSession (debugAdapter.ts:242)
    at t.stopServer (rawDebugSession.ts:501)
    at rawDebugSession.ts:492
    at Object.g [as _notify] (winjs.base.js:1191)
    at Object.enter (winjs.base.js:867)
    at n.Class.derive._oncancel._run (winjs.base.js:1089)
    at n.Class.derive._oncancel._completed (winjs.base.js:1057)
    at rawDebugSession.ts:226
    at t.e.acceptMessage (debugAdapter.ts:117)
    at t.handleData (debugAdapter.ts:189)
    at Socket.<anonymous> (debugAdapter.ts:155)
    at emitOne (events.js:116)
    at Socket.emit (events.js:211)
    at addChunk (_stream_readable.js:263)
    at readableAddChunk (_stream_readable.js:250)
    at Socket.Readable.push (_stream_readable.js:208)
    at TCP.onread (net.js:594)
```

**Comment by DanTup** [comment]
There's an open VS Code issue about this at https://github.com/Microsoft/vscode/issues/56700 (also raised by a Flutter user). Not sure if this is our bug or theirs (nobody else seems to have reported it, suggesting it might be only affecting Flutter even if it's not our bug).

**Comment by DanTup** [comment]
Please try the fix out in v2.18.0 Beta 1:

https://github.com/Dart-Code/Dart-Code/releases/tag/v2.18.0-beta.1
